api2-war/src/main/java/eu/europeana/api2/v2/web/controller/ContentReuseFrameworkController.java

```diff
@@ -42,7 +71,99 @@
     @Resource
     private ControllerUtils controllerUtils;
 
-   
+    @RequestMapping(value = "/v2/metadata-by-url.json", method = RequestMethod.GET,
+                           produces = MediaType.APPLICATION_JSON_VALUE)
+    public ModelAndView metadataByUrl(
+                                             @RequestParam(value = "url", required = true) String url,
+                                             @RequestParam(value = "wskey", required = true) String wskey,
+                                             @RequestParam(value = "callback", required = false) String callback,
+                                             HttpServletRequest request,
+                                             HttpServletResponse response) {
+        long t0 = System.currentTimeMillis();
+        controllerUtils.addResponseHeaders(response);
+        LimitResponse limitResponse = null;
+        try {
+            limitResponse = controllerUtils.checkLimit(wskey, request.getRequestURL().toString(),
+                                                       "record.json", RecordType.OBJECT, null);
+        } catch (ApiLimitException e) {
+            response.setStatus(e.getHttpStatus());
+            return JsonUtils.toJson(new ApiError(e), callback);
+        }
+
+        CrfMetadataResult result = new CrfMetadataResult(wskey, "metadata-by-url.json", limitResponse.getRequestNumber());
+        SourceDocumentReferenceMetaInfo info = crfService.getMetadata(url);
+        if (info != null) {
+            result.imageMetaInfo = info.getImageMetaInfo();
+        }
+        result.statsDuration = (System.currentTimeMillis() - t0);
+        return JsonUtils.toJson(result, callback);
+    }
+
+    @RequestMapping(value = "/v2/thumbnail-by-url.json", method = RequestMethod.GET)
+    public ResponseEntity<byte[]> thumbnailByUrl(
+                                                        @RequestParam(value = "uri", required = true) String url,
+                                                        @RequestParam(value = "size", required = false, defaultValue = "FULL_DOC") String size,
+                                                        @RequestParam(value = "type", required = false, defaultValue = "IMAGE") String type,
+                                                        @RequestParam(value = "callback", required = false) String callback,
+                                                        HttpServletRequest request,
+                                                        HttpServletResponse response) {
+        controllerUtils.addResponseHeaders(response);
+
+        log.info("Thumbnail");
+
+        String sufix = "";
+        log.info("Size: " + size);
```

Comment:
removed the logs altogether as it doesn't make sense. all that data is anyhow visible in the API